### ISSUE TITLE
BinaryFormatter serialization obsolete

### ISSUE THREAD

**Issue by ooooolivia** [dotnet/performance#1402]
```'BinaryFormatter.Serialize(Stream, object)' is obsolete: 'BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.'``` 

https://github.com/dotnet/runtime/commit/324c6c7f1240b6cf92e1f3215cf35e9206ec100c

There's an update in the runtime that obsoleted this method and action should be taken to remove the error.

**Comment by ooooolivia** [comment]
@GrabYourPitchforks The action we are temporarily taking is to suppress the warning but still test this microbenchmark Any suggestions?

**Comment by alexcovington** [comment]
Also experiencing this error, currently is blocking my workflow. Is there a workaround I can use in the meantime?

**Comment by GrabYourPitchforks** [comment]
@alexcovington The suppressions are already checked in at https://github.com/dotnet/performance/pull/1403. Consumers of this repo should be unblocked.

This repo is essentially a test project. Our own test projects suppress this obsoletion because we need to be able to test the runtime functionality. So it would be appropriate for this repo to do the same.

**Comment by alexcovington** [comment]
@GrabYourPitchforks Thanks. I'll pull and update my local copy.

**Comment by danmosemsft** [comment]
Seems this can be closed now then @ooooolivia ?

**Comment by adamsitnik** [comment]
> So it would be appropriate for this repo to do the same.

**Comment by adamsitnik** [comment]
> So it would be appropriate for this repo to do the same.

Have we considered the removal of the `BinaryFormatter` benchmarks? If the type is obsolete and we want our users to stop using it, perhaps we should stop using it ourselves?

**Comment by GrabYourPitchforks** [comment]
I think it's still worthwhile to run minimal benchmarks. We say that even though we're no longer fixing issues with it, we'll consider serious perf regressions. But I wouldn't invest in any _new_ benchmarks in this area.

**Comment by ooooolivia** [comment]
> I think it's still worthwhile to run minimal benchmarks. We say that even though we're no longer fixing issues with it, we'll consider serious perf regressions. But I wouldn't invest in any _new_ benchmarks in this area.

Our intention was to still test it as long as there are users using it. However, my second thought is, even if we catch severe regression at some point in the future, we would just tell the user this method is obsolete and won't do any performance improvement work for an obsolete method ( just my assumption, correct me if I'm wrong).

**Comment by danmosemsft** [comment]
If we seriously regressed its performance we would potentially service for that. So I think we should protect perf for a bit longer.

**Comment by GrabYourPitchforks** [comment]
We would potentially service severe performance regressions. It's called out [in the obsoletion docs](https://github.com/dotnet/designs/blob/master/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#all-outstanding-binaryformatter-related-issues-resolved-wont-fix-net-5) as something we'd consider.
